database/scrapers/departments.ts

```diff
@@ -68,46 +67,34 @@ export default class DepartmentsScraper extends BaseScraperModule {
     const [departmentsData, fieldOfStudyData, departmentLinkData] =
       (await Promise.all(
         this.directusSchemas.map((schema) =>
-          this.semaphore.runTask(() =>
-            this.fetchJSON(
-              `https://admin.topwr.solvro.pl/items/${schema}?limit=-1`,
-              schema,
-            ),
-          ),
+          this.semaphore
+            .runTask(() =>
+              this.fetchJSON(
+                `https://admin.topwr.solvro.pl/items/${schema}?limit=-1`,
+                schema,
+              ),
+            )
+            .then((response) => {
+              assertResponseStructure(response);
+              return response;
+            }),
         ),
       )) as [
```

Comment:
you need to add error context here, else it will be impossible to tell which schema failed the check